tests/ffuncs/test_ffunc_count.py

```diff
@@ -89,95 +104,168 @@ def test_single_arr_with_nan_workflow(self):
-
-        counts, validity = ccube([idx1]).count(weights=wt, return_missing_as=(0, False))
-        assert counts.tolist() == [0, (0.25 + 0.99)]
-        assert validity.tolist() == [False, True]
-
-        counts, validity = ccube([idx1]).count(
-            weights=wt, ignore_missing=True, return_missing_as=(0, False)
-        )
-        assert counts.tolist() == [(0.3 + 1.0), (0.25 + 0.99)]
-        assert validity.tolist() == [True, True]
-
-        counts = ccube([idx1, idx2]).count(weights=wt)
-        assert arr_eq(counts, [[float("nan"), 1.0], [0.99, 0.25]])
-
-        counts, validity = ccube([idx1, idx2]).count(
-            weights=wt, return_missing_as=(0, False)
-        )
-        assert counts.tolist() == [[0, 1.0], [0.99, 0.25]]
-        assert validity.tolist() == [[False, True], [True, True]]
-
-        counts = ccube([iindex({(2,): [0, 2]}, 0, (5,))]).count()
-        # Force cells that had no inputs to 0 by using return_missing_as=0.
-        assert arr_eq(counts, [3, float("nan"), 2])
-        counts = ccube([iindex({(2,): [0, 2]}, 0, (5,))]).count(return_missing_as=0)
-        # Force cells that had no inputs to 0 by using return_missing_as=0.
-        assert arr_eq(counts, [3, 0, 2])
+        with compare_ccube_to_xcube():
+            counts = ccube([idx1]).count(weights=None)
+            assert arr_eq(counts, [3, 2])
+
+            counts = ccube([idx1]).count(weights=[0.25, 0.3, 0.99, 1.0, 0.5])
+            assert arr_eq(counts, [1.8, 1.24])
+
+            counts = ccube([idx1]).count(weights=[0.25, 0.3, 0.99, 1.0, float("nan")])
+            assert arr_eq(counts, [float("nan"), 1.24])
+
+            counts = ccube([idx1]).count(
+                weights=[0.25, 0.3, 0.99, 1.0, float("nan")], ignore_missing=True
+            )
+            assert arr_eq(counts, [1.3, 1.24])
+
+            counts = ccube([idx1]).count(weights=0.5)
+            assert arr_eq(counts, [1.5, 1.0])
+
+            counts = ccube([idx1]).count(weights=float("nan"))
+            assert arr_eq(counts, [float("nan"), float("nan")])
+
+            counts = ccube([idx1]).count(weights=float("nan"), ignore_missing=True)
+            assert arr_eq(counts, [float("nan"), float("nan")])
+
+
+class TestFfuncCountMissingness:
+    # Make sure each combination of these options works properly:
+    #  * ignore_missing = True/False
+    #  * return_missing_as = NaN/(<sentinel value>, False)
+    #  * at least one output cell which has no inputs
+    #  * [count() takes no fact variable]
+    #  * weights variable (with missings) as:
+    #    * single array with NaN
+    #    * no NaN, but separate "validity" array
+    #    * includes NaN, but separate "validity" array
+    idx_with_empty_cell = iindex({(1,): [3]}, 0, (5,))  # [0, 0, 0, 1, 0]
+
+    dirty_weights = [9.0, 9.0, float("nan"), 9.0, 9.0]
+    clean_weights = [9.0, 9.0, MAXFLOAT, 9.0, 9.0]
+    weights_validity = [True, True, False, True, True]
+
+    args = [
```

Comment:
Although these _are_ args to the test parameterize decorator, I tend to think of them as _parameters_, for the tests. So maybe consider renaming (I know it's a lot given the PR size). Maybe something like `weights_test_parameters`, or `weights_parameters`.

And then in other files where you have the same structure you'd use a different name, like `factvar_and_weights_parameters`. This would improve the readability of the decorators a tiny bit.